src/System.Device.Gpio/System/Device/Gpio/ExtendedPinMode.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class ExtendedPinMode : IEquatable<ExtendedPinMode>
+    {
+        private readonly string _modeName;
+
+        /// <summary>
+        /// The mode is not known
+        /// </summary>
+        public static readonly ExtendedPinMode Unknown = new ExtendedPinMode("Unknown");
+
+        /// <summary>
+        /// Creates a new mode.
+        /// </summary>
+        /// <param name="modeName">Common name of the mode</param>
+        protected ExtendedPinMode(string modeName)
```

Comment:
I'd suggest to do this argumentless and make implementation choose if they want to use string or something else to get a name (i.e. in RaspberryPiExtendedMode you don't really need any string)

Reply:
I know that the string is not absolutely necessary, but it helps in keeping the things testable. Also, otherwise the instance wouldn't necessarily have any identity at all and comparing them was impossible.